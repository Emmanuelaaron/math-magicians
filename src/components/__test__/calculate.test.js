import calculate from '../../logic/calculate';

let myObj = {};
let myObj1 = {};
let myObj2 = {};
describe('Calculator testing', () => {
  beforeEach(() => {
    myObj = {
      total: 3,
      next: 1,
      operation: '+',
    };
    myObj1 = {
      total: 3,
      next: '0',
      operation: '+',
    };
    myObj2 = {
      total: 3,
      next: null,
      operation: '+',
    };
  });

  afterEach(() => {
    myObj = {};
    myObj1 = {};
    myObj2 = {};
  });
  it('returns 0 when AC button is clicked', () => {
    const res = calculate(myObj, 'AC');
    expect(res).toMatchSnapshot({
      total: null,
      next: null,
      operation: null,
    });
  });

  it('returns the result if "=" is clicked', () => {
    const res = calculate(myObj, '=');
    expect(res).toMatchSnapshot({
      total: '4',
      next: null,
      operation: null,
    });
  });

  test("result of '+/-'", () => {
    const res = calculate(myObj, '+/-');
    expect(res).toMatchSnapshot({
      total: 3,
      next: '-1',
      operation: '+',
    });
  });

  test('if button is a number', () => {
    const res = calculate(myObj, '9');
    expect(res).toMatchSnapshot({
      total: 3,
      next: '19',
      operation: '+',
    });
  });

  test('if button is a 0 and next is 0 too', () => {
    const res = calculate(myObj1, '0');
    expect(res).toMatchSnapshot({});
  });

  test('if button is a 0 and next is 0 too', () => {
    const res = calculate(myObj2, '8');
    expect(res).toMatchSnapshot({});
  });

  test('if button is a .', () => {
    const res = calculate(myObj1, '.');
    expect(res).toMatchSnapshot({});
  });
});
