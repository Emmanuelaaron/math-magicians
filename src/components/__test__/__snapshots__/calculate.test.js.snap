// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator testing if button is a . 1`] = `
Object {
  "next": "0.",
  "operation": "+",
  "total": 3,
}
`;

exports[`Calculator testing if button is a 0 and next is 0 too 1`] = `Object {}`;

exports[`Calculator testing if button is a 0 and next is 0 too 2`] = `
Object {
  "next": "8",
  "operation": "+",
  "total": 3,
}
`;

exports[`Calculator testing if button is a number 1`] = `
Object {
  "next": "19",
  "operation": "+",
  "total": 3,
}
`;

exports[`Calculator testing result of '+/-' 1`] = `
Object {
  "next": "-1",
  "operation": "+",
  "total": 3,
}
`;

exports[`Calculator testing returns 0 when AC button is clicked 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": null,
}
`;

exports[`Calculator testing returns the result if "=" is clicked 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": "4",
}
`;
